packages/shared/src/components/cards/ActionsButtons/UpvoteButtonIcon.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { feature } from '../../../lib/featureManagement';
+import { UpvoteIcon } from '../../icons';
+import { IconProps, IconSize } from '../../Icon';
+import { userPrefersReducedMotions } from '../../../styles/media';
+import { useConditionalFeature, useMedia } from '../../../hooks';
+import styles from './ActionButtons.module.css';
+
+type AnimatedButtonIconProps = IconProps & { userClicked?: boolean };
+
+const arrows = Array.from({ length: 5 }, (_, i) => i + 1);
+
+export const UpvoteButtonIcon = React.memo(function UpvoteButtonIconComp(
+  props: AnimatedButtonIconProps,
+): ReactElement {
+  const { secondary: isUpvoteActive, userClicked } = props;
+
+  const haveUserPrefersReducedMotions = useMedia(
+    [userPrefersReducedMotions.replace('@media', '')],
+    [false],
+    false,
+    false,
+  );
+
+  const { value: isAnimatedVersion } = useConditionalFeature({
+    feature: feature.animatedUpvote,
+    shouldEvaluate: !haveUserPrefersReducedMotions,
```

Comment:
@nensidosari not 100% sure on what the idea was but don't we want the evaluation to include these?
```&& userClicked && isUpvoteActive```

Reply:
well yes if we want to enrol only users who upvoted from when we roll this experiment. It's quite a specific enrolment, we can check with Ido

Reply:
https://github.com/dailydotdev/apps/pull/3318#discussion_r1671676473